tirocks-sys/src/lib.rs

```diff
@@ -13,20 +13,196 @@ mod bindings {
     include!(env!("BINDING_PATH"));
 }
 
+use std::{ffi::CStr, str::Utf8Error};
+
 pub use bindings::*;
 
-impl crocksdb_slice_t {
-    pub fn to_bytes(&self) -> &[u8] {
-        unsafe { std::slice::from_raw_parts(self.data as _, self.size) }
+/// Convert a rust slice to rocksdb slice.
+///
+/// # Safety
+///
+/// Caller should guarantee the data lives as long as the returned slice.
+// Because the layout of rocksdb_Slice is exact the same as &[u8], this function
+// will be optimized away by compiler.
+#[inline]
+pub unsafe fn r(src: &[u8]) -> rocksdb_Slice {
+    rocksdb_Slice {
+        data_: src.as_ptr() as _,
+        size_: src.len(),
     }
 }
 
+/// Convert a rocksdb slice to rust slice.
+///
+/// # Safety
+///
+/// Caller should guarantee the data lives as long as the returned slice.
+// Because the layout of rocksdb_Slice is exact the same as &[u8], this function
+// will be optimized away by compiler.
+#[inline]
+pub unsafe fn s<'a>(src: rocksdb_Slice) -> &'a [u8] {
```

Comment:
give it a meaningful name?

Reply:
That will make a long function name. Giving it maybe a frequently used function. `s` stands for `slice`, `r` stands for `rocksdb`.